frontend/app/components/auth/components/oauth.tsx

```diff
@@ -1,29 +1,33 @@
 import { h, JSX } from 'preact';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import clsx from 'clsx';
 import { useIntl } from 'react-intl';
 
 import type { OAuthProvider } from 'common/types';
 import { siteId } from 'common/settings';
 import { useTheme } from 'hooks/useTheme';
-import { setUser } from 'store/user/actions';
+import { setUser, setTelegramParams } from 'store/user/actions';
+import { StoreState } from 'store';
 
 import { messages } from 'components/auth/auth.messsages';
 
 import { oauthSignin } from './oauth.api';
 import { BASE_URL } from 'common/constants.config';
 import { getButtonVariant, getProviderData } from './oauth.utils';
 import styles from './oauth.module.css';
+import { getTelegramSigninParams } from '../auth.api';
 
 const location = encodeURIComponent(`${window.location.origin}${window.location.pathname}?selfClose`);
 
 type Props = {
   providers: OAuthProvider[];
+  toggleTelegram?: (showTelegram: boolean) => void;
 };
 
-export function OAuth({ providers }: Props) {
+export function OAuth({ providers, toggleTelegram }: Props) {
   const intl = useIntl();
   const dispath = useDispatch();
+  const telegramParams = useSelector((s: StoreState) => s.telegramParams);
```

Comment:
I don't see any reason for redux here. We handle all of the logic inside `Auth` component.

I would suggest adding `onClickProvider`  callback for `OAuth` and handling this case in the parent component.

Reply:
What is the reason? We already have redux used in this file prior to my changes.

Reply:
The reason is we shouldn't put local state into global store. It's component specific logic and it doesn't affect application. Another reason is to keep it simple and don't write more code than we need. Now logic is scattered between store and this component.

For `user` we put it to the global store and then use in other components that placed in other places.